pkg/discovery/metrics/metric_sink.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/turbostore"
+	"reflect"
+)
+
+type EntityMetricSink struct {
+	data *turbostore.Cache
+}
+
+func NewEntityMetricSink() *EntityMetricSink {
+	return &EntityMetricSink{
+		data: turbostore.NewCache(),
+	}
+}
+
+// Add one or more metric entries to sink.
+func (s *EntityMetricSink) AddNewMetricEntries(metric ...Metric) {
```

Comment:
Performance concerns:
      each s.data.Add() requires a Lock. 

It would be better to implement a AddList() interface to add bulk of items with one lock request.